### PR TITLE
Bug 2089402: Validate Image labels correctly

### PR DESCRIPTION
**Pull request by jkhelil**

### CODE REVIEW

File: pkg/build/apis/build/validation/validation_test.go [hunk 1]
```diff
@@ -2564,7 +2564,7 @@ func TestValidateCommonSpecSuccess(t *testing.T) {
 					ImageLabels: []buildapi.ImageLabel{
 						{
 							Name:  "key",
-							Value: "value )(*&",
+							Value: "value-_.v",
```
Comment (coreydaley) [inline pkg/build/apis/build/validation/validation_test.go:2567]: Why did this need to change?

File: pkg/build/apis/build/validation/validation_test.go [hunk 2]
```diff
@@ -3944,3 +3944,83 @@ func TestValidateBuildVolumes(t *testing.T) {
...
+				{
+					Name:  "key1",
+					Value: "value1",
+				},
+				{
+					Name:  "prefix/key2",
+					Value: "value2",
+				},
+			},
+		},
```
Comment (coreydaley) [inline pkg/build/apis/build/validation/validation_test.go:4003]: Missing the explicit err fields.

Comment (jkhelil) [reply pkg/build/apis/build/validation/validation_test.go:4003]: we shouldnt have errors here, it is passing a test

File: pkg/build/apis/build/validation/validation_test.go (hunk 1 above)
Comment (coreydaley) [reply pkg/build/apis/build/validation/validation_test.go:2567]: My concern is that if we make this more restrictive, all of the sudden we will be generating errors for existing labels that include those characters (if any exist)

File: pkg/build/apis/build/validation/validation.go [hunk 3]
```diff
@@ -735,10 +735,10 @@ func ValidateImageLabels(labels []buildapi.ImageLabel, fldPath *field.Path) (all
...
 		if len(lbl.Name) == 0 {
 			allErrs = append(allErrs, field.Required(idxPath.Child("name"), ""))
-			continue
-		}
-		for _, msg := range kvalidation.IsConfigMapKey(lbl.Name) {
-			allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), lbl.Name, msg))
+		} else if errs := kvalidation.IsQualifiedName(lbl.Name); len(errs) != 0 {
+			allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), lbl.Name, strings.Join(errs, "; ")))
+		} else if errs := kvalidation.IsValidLabelValue(lbl.Value); len(errs) != 0 {
+			allErrs = append(allErrs, field.Invalid(idxPath.Child("value"), lbl.Value, strings.Join(errs, "; ")))
```
Comment (coreydaley) [inline pkg/build/apis/build/validation/validation.go:741]: We were not previously checking the value, only the name, so let's not add the checking of the value, that should fix the `making the image label value more restrictive` issue and you should be able to revert the other test change.  If we get a bug about image values, then we will address it, but we probably won't be able to backport that change.

File: pkg/build/apis/build/validation/validation_test.go (hunk 1 above)
Comment (coreydaley) [reply pkg/build/apis/build/validation/validation_test.go:2567]: Revert this and see above comment about having added the `IsValidLabelValue`.

File: pkg/build/apis/build/validation/validation_test.go (hunk 2 above)
Comment (jkhelil) [reply pkg/build/apis/build/validation/validation_test.go]: done

File: pkg/build/apis/build/validation/validation.go (hunk 3 above)
Comment (jkhelil) [reply pkg/build/apis/build/validation/validation.go]: Removed

File: pkg/build/apis/build/validation/validation_test.go (hunk 1 above)
Comment (jkhelil) [reply pkg/build/apis/build/validation/validation_test.go]: fixed, I removed the check for label values

**Comment by bparees** [comment]
/approve

@coreydaley please work with the api-server team to get yourself added to https://github.com/openshift/openshift-apiserver/blob/master/pkg/build/OWNERS (and perhaps remove the old/stale entries)

**Comment by coreydaley** [comment]
@bparees I have opened https://github.com/openshift/openshift-apiserver/pull/294

**Pull request merged**